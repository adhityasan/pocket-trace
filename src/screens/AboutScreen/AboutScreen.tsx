import { SafeAreaView, StatusBar, Text, TouchableOpacity, View } from "react-native";

import { RootStackProps } from "@@types/navigations/root";
import EXTRA from "@services/constants/extra";
import { SCREEN_NAMES } from "@services/constants/screen";
import globalStyles from "@styles/_global";

type AboutScreenParamsType = RootStackProps<SCREEN_NAMES.ABOUT>;

const AboutScreen = ({ navigation }: AboutScreenParamsType) => (
  <SafeAreaView style={[globalStyles.screenView, { backgroundColor: "#919ff0" }]}>
    <StatusBar barStyle="dark-content" backgroundColor="#919ff0" />
    <View>
      <Text style={[globalStyles.titleText, { color: "#232323" }]}>About Screen</Text>
      <Text>{JSON.stringify(EXTRA, null, 2)}</Text>
      <TouchableOpacity onPress={() => navigation.navigate(SCREEN_NAMES.HOME)}>
        <Text style={{ color: "#434343", textAlign: "center" }}>back to home screen ?</Text>
      </TouchableOpacity>
    </View>
  </SafeAreaView>
);

export default AboutScreen;
