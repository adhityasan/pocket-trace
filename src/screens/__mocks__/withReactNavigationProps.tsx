import React from "react";

export const createTestScreenProps = (props?: object) => ({
  navigation: {
    state: { params: {} },
    dispatch: jest.fn(),
    goBack: jest.fn(),
    dismiss: jest.fn(),
    navigate: jest.fn(),
    openDrawer: jest.fn(),
    closeDrawer: jest.fn(),
    toggleDrawer: jest.fn(),
    getParam: jest.fn(),
    setParams: jest.fn(),
    addListener: jest.fn(),
    push: jest.fn(),
    replace: jest.fn(),
    pop: jest.fn(),
    popToTop: jest.fn(),
    isFocused: jest.fn()
  },
  ...props
});

const withReactNavigationProps = (Component: any) => {
  const navigationProps = createTestScreenProps();

  return <Component {...navigationProps} />;
};

export default withReactNavigationProps;
