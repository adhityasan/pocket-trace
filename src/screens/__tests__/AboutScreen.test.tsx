import { create, act } from "react-test-renderer";

import AboutScreen from "@screens/AboutScreen";
import withReactNavigationProps from "@screens/__mocks__/withReactNavigationProps";

describe("screens/AboutScreen", () => {
  it("should match snapshot", () => {
    const withNavigationProps = withReactNavigationProps(AboutScreen);
    const result = create(withNavigationProps).toJSON();
    act(() => {
      expect(result).toMatchSnapshot();
    });
  });
});
