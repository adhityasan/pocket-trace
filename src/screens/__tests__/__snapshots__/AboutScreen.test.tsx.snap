// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`screens/AboutScreen should match snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      undefined,
      Object {
        "backgroundColor": "#919ff0",
      },
    ]
  }
>
  <View>
    <Text
      style={
        Array [
          Object {
            "fontSize": 30,
            "fontWeight": "bold",
            "textAlign": "center",
          },
          Object {
            "color": "#232323",
          },
        ]
      }
    >
      About Screen
    </Text>
    <Text>
      {
  "APP_NAME": "Pocket Trace",
  "APP_ENV": "development",
  "APP_API_BASE_URL": "http://localhost:3000"
}
    </Text>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#434343",
            "textAlign": "center",
          }
        }
      >
        back to home screen ?
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
